WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java

```diff
@@ -94,23 +97,31 @@ private void refreshContent() {
     }
 
     private void downloadExternalMedia() {
-        if (Intent.ACTION_SEND_MULTIPLE.equals(getIntent().getAction())) {
-            ArrayList<Uri> externalUris = getIntent().getParcelableArrayListExtra((Intent.EXTRA_STREAM));
-            for (Uri uri : externalUris) {
-                if (uri != null && isAllowedMediaType(uri)) {
-                    mLocalMediaUris.add(MediaUtils.downloadExternalMedia(this, uri));
+        try {
+            if (Intent.ACTION_SEND_MULTIPLE.equals(getIntent().getAction())) {
+                ArrayList<Uri> externalUris = getIntent().getParcelableArrayListExtra((Intent.EXTRA_STREAM));
+                for (Uri uri : externalUris) {
+                    if (uri != null && isAllowedMediaType(uri)) {
+                        mLocalMediaUris.add(MediaUtils.downloadExternalMedia(this, uri));
+                    }
+                }
+            } else if (Intent.ACTION_SEND.equals(getIntent().getAction())) {
+                Uri externalUri = getIntent().getParcelableExtra(Intent.EXTRA_STREAM);
+                if (externalUri != null && isAllowedMediaType(externalUri)) {
+                    mLocalMediaUris.add(MediaUtils.downloadExternalMedia(this, externalUri));
                 }
             }
-        } else if (Intent.ACTION_SEND.equals(getIntent().getAction())) {
-            Uri externalUri = getIntent().getParcelableExtra(Intent.EXTRA_STREAM);
-            if (externalUri != null && isAllowedMediaType(externalUri)) {
-                mLocalMediaUris.add(MediaUtils.downloadExternalMedia(this, externalUri));
-            }
+        } catch (Exception e) {
+            AppLog.e(T.MEDIA, "ShareIntentReceiver failed to download media ", e);
         }
     }
 
     private boolean isAllowedMediaType(@NonNull Uri uri) {
         String filePath = MediaUtils.getRealPathFromURI(this, uri);
+        // For cases when getRealPathFromURI returns an empty string
+        if (TextUtils.isEmpty(filePath)) {
+            filePath = String.valueOf(uri);
+        }
```

Comment:
I wonder why the URI is not properly parsed. I debugged the code and found that it's caused by [this line](https://github.com/wordpress-mobile/WordPress-Utils-Android/blob/c0c7e4a342fb7ffd625fc8962d129293302d9731/WordPressUtils/src/main/java/org/wordpress/android/util/MediaUtils.java#L440) returning `false`. My presumption is that for URIs like the following it should work:
- `content://media/external/images/media/1000000246` (media file from Gallery app)
- `content://com.adobe.lrmobile.provider/myimages/20231013_1723402258340274679050296%20(17).jpg` (media file from Adobe Lightroom app)

Might be related to the Android version, although I haven't found any references in the documentation.

[This code was introduced in 2017](https://github.com/wordpress-mobile/WordPress-Utils-Android/commit/a44209420022fb13265e6ca83545f2b662f8c6b8), so it might be a long shot, but I wonder @daniloercoli as author, if you could provide any info about this logic. Thanks for the help 🙇 !

Reply:
It might not be critical, but the fact that [we extract the path from the URI](https://github.com/wordpress-mobile/WordPress-Utils-Android/blob/c0c7e4a342fb7ffd625fc8962d129293302d9731/WordPressUtils/src/main/java/org/wordpress/android/util/MediaUtils.java#L438-L508) makes me hesitant about whether we can simply convert it to `string`. Not sure if this might introduce undesired side-effects.

Reply:
I realize that the function `isAllowedMediaType` was introduced recently in https://github.com/wordpress-mobile/WordPress-Android/pull/19754:

https://github.com/wordpress-mobile/WordPress-Android/blob/0c988413de6353275648fa84df9be9d8fd4fb365/WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java#L119-L126

This reduces my concern about providing a fallback in case the URI can't be processed, as mentioned in https://github.com/wordpress-mobile/WordPress-Android/pull/20138#discussion_r1481890026. Hence, I think we can proceed with this approach.

Reply:
Yeah I think the whole code that handles this should be revisited with the app permissions logic as well, which is something that was changed recently. The sharing media behavior is also different between OS versions and apps, so definitely worth looking into it to fix issues like this one https://github.com/wordpress-mobile/WordPress-Android/issues/20140